### DISCUSSION TITLE
FilesService with a failed transaction does not remove the file from the storage

### DISCUSSION THREAD

**Discussion by Enhed** [directus/directus#16551] [Feature Requests]
### Describe the Bug

After a transaction has fallen, the interface and the database obviously do not have such a file, but physically the file stays in the storage.  

I know that the transaction affects exactly the record in the database, but it would still be great if the file was deleted, otherwise you could get lost files that take up space

### To Reproduce

i write a simple enpoint to reproduce that:  

```ts
import {defineEndpoint} from "@directus/extensions-sdk";
import {Knex} from "knex";
import {FilesService, InvalidPayloadException} from "directus";

export default defineEndpoint((router, { getSchema, database, emitter }) => {
    router.get('/', async (req, res, next) => {
        try {
            await database.transaction(async (trx: Knex) => {
                const service = new FilesService({
                    schema: await getSchema(),
                    knex: trx
                })

                const key = await service.importOne('url to img', {})
                console.log('loaded file with key', key)

                throw new InvalidPayloadException('test error')
            })
        }
        catch (e) {
            next(e)
        }
    })
})

**Comment by azrikahar** [comment]
As there should be access to the file ID (`key` variable in your example), I think it should be possible to declare `key` variable outside of the try/catch, then check whether it's assigned a value inside the catch block. If it does, delete the file first before calling `next()` 🤔 Not 100% sure but feels like that can be an alternative if there aren't any issue with that approach.

Moving this to discussion in the meantime as this feels like a rather specific use case (I could be very wrong here!) and to both gauge the community interest while also gather more thoughts before we can turn this into an actionable item 👍

--- 

EDIT:

> this feels like a rather specific use case (I could be very wrong here!)

Schema applies does currently wrap collections/fields/relations service within a transaction, so I stand corrected! 😄

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
